models/database_test.go

```diff
@@ -152,7 +152,27 @@ func TestDatabaseChecks(t *testing.T) {
 	})
 
 	t.Run("Services", func(t *testing.T) {
-		t.Skip("TODO")
+		db := testdb.Open(t, models.SkipFixtures, nil)
+		defer func() {
+			require.NoError(t, db.Close())
+		}()
+		var err error
+		now := models.Now()
+		_, err = db.Exec(
+			"INSERT INTO nodes (node_id, node_type, node_name, distro, node_model, az, address, created_at, updated_at) "+
+				"VALUES ('/node_id/1', 'generic', 'name', '', '', '', '', $1, $2)",
+			now, now,
+		)
+		require.NoError(t, err)
+
+		// Try to insert both address and socket
+		_, err = db.Exec(
+			"INSERT INTO services (service_id, service_type, service_name, node_id, environment, cluster, replication_set, address, port, socket, created_at, updated_at) "+
+				"VALUES ('/service_id/1', 'mysql', 'name', '/node_id/1', '', '', '', '10.10.10.10', 3306, '/var/run/mysqld/mysqld.sock', $1, $2)",
+			now, now,
+		)
+		require.Error(t, err, `pq: new row for relation "services" violates check constraint "address_socket_check"`)
+
```

Comment:
let's try with both empty

Reply:
can we also add the check that if address is not empty then port is also shouldn't be empty

Reply:
Added